### DISCUSSION TITLE
Is there any way to enable github webhook into argocd core installation?

### DISCUSSION THREAD

**Discussion by ms-choudhary** [argoproj/argo-cd#8351] [Q&A]
I've setup [argocd core installation](https://argo-cd.readthedocs.io/en/stable/operator-manual/installation/#core). I've read the instructions listed [here](https://argo-cd.readthedocs.io/en/stable/operator-manual/webhook/). But that requires argocd-server installation (which isn't installed in core). Is it possible to setup github webhooks for core? Or is it something that requires regular installation?

**Comment by blackandred** [comment]
I have the same case. I installed Argo Core to have a lightweight and smaller setup, but using git-polling instead of webhooks is not lightweight - I would like to use webhooks.

**Reply by rishabh625** [reply]
I guess not, webhook server comes as a part of api server installation and for argocd core installation api server is not included, but i agree with point that core installation should have webhook server.

**Reply by blackandred** [reply]
Is there a way to perform a repo refresh from CLI?

**Comment by rishabh625** [comment]
yeah start an repo-server with --default-cache-expiration duration as you want
https://argo-cd.readthedocs.io/en/stable/operator-manual/server-commands/argocd-repo-server/

**Comment by nielsole** [comment]
Related Issue: https://github.com/argoproj/argo-cd/issues/9378
